src/components/public-name-tag/controllers/public-name-tag.controller.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  Body,
+  CacheInterceptor,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  Post,
+  Put,
+  Query,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiForbiddenResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiQuery,
+  ApiResponse,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import { PublicNameTagService } from '../services/public-name-tag.service';
+import {
+  AkcLogger,
+  BaseApiResponse,
+  MESSAGES,
+  ReqContext,
+  RequestContext,
+  USER_ROLE,
+} from '../../../shared';
+import { PublicNameTagParamsDto } from '../dtos/public-name-tag-params.dto';
+import { StorePublicNameTagParamsDto } from '../dtos/store-public-name-tag-params.dto';
+import { RoleGuard } from '../../../auth/role/roles.guard';
+import { Roles } from '../../../auth/role/roles.decorator';
+import { JwtAuthGuard } from '../../../auth/jwt/jwt-auth.guard';
+import { PublicNameTag } from '../../../shared/entities/name-tag.entity';
+import { GetPublicNameTagResult } from '../dtos/get-public-name-tag-result.dto';
+import { GetPublicNameTagAdminResult } from '../dtos/get-public-name-tag-admin.dto';
+import { bech32 } from 'bech32';
+
+@Controller()
+@ApiTags('public-name-tag')
+@ApiUnauthorizedResponse({
+  description: MESSAGES.ERROR.NOT_PERMISSION,
+})
+@ApiForbiddenResponse({
+  description: MESSAGES.ERROR.NOT_PERMISSION,
+})
+@ApiBadRequestResponse({
+  description: MESSAGES.ERROR.BAD_REQUEST,
+})
+export class PublicNameTagController {
+  constructor(
+    private readonly nameTagService: PublicNameTagService,
+    private readonly logger: AkcLogger,
+  ) {
+    this.logger.setContext(PublicNameTagController.name);
+  }
+
+  @Get('admin/public-name-tag')
+  // @UseGuards(JwtAuthGuard, RoleGuard)
+  // @Roles(USER_ROLE.ADMIN)
+  @ApiBearerAuth()
+  @ApiOperation({ summary: 'Get list name tag' })
+  @HttpCode(HttpStatus.OK)
+  @ApiOkResponse({ type: GetPublicNameTagAdminResult })
+  async getNameTags(
+    @ReqContext() ctx: RequestContext,
+    @Query() request: PublicNameTagParamsDto,
+  ): Promise<BaseApiResponse<PublicNameTag[]>> {
+    this.logger.log(ctx, `${this.getNameTags.name} was called!`);
+    const { result, count } = await this.nameTagService.getPublicNameTags(
+      ctx,
+      request,
+    );
+
+    return { data: result, meta: { count } };
+  }
+
+  @Get('admin/public-name-tag:id')
```

Comment:
```suggestion
  @Get('admin/public-name-tag/:id')
```
Anh thấy dạng path param như này đẹp hơn chứ nhỉ.